src/util/desktophelper.cpp

```diff
@@ -0,0 +1,169 @@
+        const QUrl fileurl = QUrl::fromLocalFile(path);
+        const QStringList args(fileurl.toString());
+        QDBusMessage msg = QDBusMessage::createMethodCall(
+                "org.freedesktop.FileManager1",
+                "/org/freedesktop/FileManager1",
+                "org.freedesktop.FileManager1",
+                "ShowItems");
+        msg << args << "";
+        const QDBusMessage response = QDBusConnection::sessionBus().call(msg);
+        const bool success = (response.type() != QDBusMessage::MessageType::ErrorMessage);
+        return success;
+    }
+
+    bool selectInXfce(const QString& path) {
+        const QString folder = QFileInfo(path).dir().absolutePath();
+        const QString filename = QFileInfo(path).fileName();
+
+        QDBusMessage msg = QDBusMessage::createMethodCall(
+                "org.xfce.FileManager",
+                "/org/xfce/FileManager",
+                "org.xfce.FileManager",
+                "DisplayFolderAndSelect");
+        msg << folder << filename << "" << "";
+        qDebug() << "Calling:" << msg;
+        const QDBusMessage response = QDBusConnection::sessionBus().call(msg);
+        const bool success = (response.type() != QDBusMessage::MessageType::ErrorMessage);
+        return success;
+    }
+#endif
+
+} // anonymous namespace
+
+
+namespace mixxx {
+
+void DesktopHelper::openInFileBrowser(const QStringList& paths) {
+    if (sSelectInFileBrowserCommand.isNull()) {
+        sSelectInFileBrowserCommand = getSelectInFileBrowserCommand();
+    }
+    QSet<QString> openedDirs;
+    QString dirPath;
+    for (const auto& path: paths) {
+        dirPath = removeChildDir(path);
+        // First try to select the file in file browser
+
+#ifdef Q_OS_LINUX
+        if (sSelectInFileBrowserCommand == kSelectInFreedesktop &&
+                QFile::exists(path)) {
+            if (!openedDirs.contains(dirPath)) {
+                if (selectInFreedesktop(path)) {
+                    openedDirs.insert(dirPath);
+                    break;
+                } else {
+                    qDebug() << "Select File via Freedesktop DBus interface failed";
+                }
+            }
+        }
+        if (sSelectInFileBrowserCommand == kSelectInXfce &&
+                QFile::exists(path)) {
+            if (!openedDirs.contains(dirPath)) {
+                if (selectInXfce(path)) {
+                    openedDirs.insert(dirPath);
+                    break;
+                } else {
+                    qDebug() << "Select file via Xfce DBus interface failed";
+                }
+            }
+        }
+#endif
+        if (sSelectInFileBrowserCommand.length() > 2 &&
+                QFile::exists(path)) {
+            if (!openedDirs.contains(dirPath)) {
+                openedDirs.insert(dirPath);
+                // special command, that supports also select the requested file
+                QString command = sSelectInFileBrowserCommand.arg(
```

Comment:
Filenames can have `"` in them -- don't these need to be escaped?

Reply:
This depends on the called application. I want to ignore it for now, because i cannot test it.
Do you have a recipe for Mac?

Reply:
I added `nautilus "%1"` handling for nautilus instead of dbus and tested with a file named `File with " in it.mp3`. Escaping with `.replace('"', "\\\"");` did not work, surprisingly. What did work was:
```c++
QString command = "nautilus";
QStringList args;
args << QDir::toNativeSeparators(path);
QProcess::startDetached(command, args);
```
It looks like QProcess with a QStringList will pass the string verbatim to the child process in `argv[x]`, with no shell in between (so no escaping is needed). I believe including the arguments in the command name will go through a shell.
https://code.woboq.org/qt5/qtbase/src/corelib/io/qprocess_unix.cpp.html#436